### ISSUE TITLE
Feature request: Support multiple packages in gRPC microservice

### ISSUE THREAD

**Issue by zry656565** [nestjs/nest#1640]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Currently, I setup a gRPC microservice, and created **several packages** in protobuf files. But it didn't work.

Cause there's only a little of documents about gRPC in Nest, so I read some Nest codes about gPRC server initialization: https://github.com/nestjs/nest/blob/bf04704e11c6a6ffc15e1188a450c77892c04792/packages/microservices/server/server-grpc.ts#L53-L74

What I found is that there's only one package that I could config in `GrpcOptions`. **ONLY ONE**. It does no good to construct my `*.proto` files in a well structure.

## Minimal reproduction of the problem with instructions

Sample codes are like below:

```ts
// grpc-server.option.ts
import { GrpcOptions, Transport } from '@nestjs/microservices'
import { join } from 'path'

export const grpcServerOptions: GrpcOptions = {
  transport: Transport.GRPC,
  options: {
    url: 'localhost:5577',
    package: 'payment',
    protoPath: join(__dirname, './proto/payment.proto'),
    loader: {
      includeDirs: [
        join(__dirname, 'proto'),
      ],
    }
  },
} 
```

```ts
// bootstrap.ts
import { NestFactory } from '@nestjs/core'

import { ApplicationModule } from './app.module'
import { grpcServerOptions } from './grpc-server.options'

export async function bootstrap() {
  const app = await NestFactory.create(ApplicationModule)
  app.connectMicroservice(grpcServerOptions)
  await app.startAllMicroservicesAsync()
}
```

## Expected behavior

I need to configure several packages when setup the server, like this:

```ts
const grpcServerOptions: GrpcOptions = {
  transport: Transport.GRPC,
  options: {
    url: 'localhost:5577',
    package: ['product', 'order'],
    protoPath: join(__dirname, './proto/payment.proto'),
    loader: {
      includeDirs: [
        join(__dirname, 'proto'),
      ],
    }
  },
} 
```

## What is the motivation / use case for changing the behavior?

I need to separate individual services into different proto files (different packages).

## Environment

<pre><code>
Nest version: 5.7.4

 
For Tooling issues:
- Node version: v10.15.0  
- Platform:  Mac 

Others:

</code></pre>

**Comment by zry656565** [comment]
## proto files

Here're more sample codes.

```
.
└── proto
    ├── common
    │   └── entity.proto
    ├── modules
    │   ├── order.proto
    │   └── product.proto
    └── payment.proto
```

```proto
// entitiy.proto
syntax = "proto3";

package entity;

enum Currency {
  USD = 0;
}

message Product {
  string id = 1;
  string slug = 2;
  string name = 3;
  string description = 4;
  Currency currency = 5;
  int32 price = 6;
}

message Order {
  string id = 1;
  string subject = 2;
  string body = 3;
  int32 totalPrice = 4;
  Currency currency = 5;
  string channel = 6;
}
```

```proto
// entitiy.proto
syntax = "proto3";

import "../common/entity.proto";

package product;

service ProductService {
  rpc FindOne (ProductRequest) returns (ProductResponse) {}
  rpc CreateOne (CreateProductRequest) returns (CreateProductResponse) {}
  rpc FindByCategory (ProductByCategoryRequest) returns (ProductByCategoryResponse) {}
}

message ProductRequest {
  string slug = 1;
}

message ProductResponse {
  entity.Product product = 3;
}

message CreateProductRequest {
  string slug = 1;
  string name = 2;
  int32 price = 3;
  Currency currency = 4;
  string description = 5;
  string categorySlug = 6;
}

message CreateProductResponse {
  string productId = 3;
}

message ProductByCategoryRequest {
  string categorySlug = 1;
}

message ProductByCategoryResponse {
  repeated entity.Product products = 1;
}
```

```proto
// order.proto
syntax = "proto3";

import "../common/entity.proto";

package order;

service OrderService {
  rpc FindOne (OrderRequest) returns (OrderResponse) {}
  rpc FindByUser (OrderByUserRequest) returns (OrderByUserResponse) {}
  rpc CreateOne (CreateOrderRequest) returns (CreateOrderResponse) {}
}

message OrderRequest {
  string id = 1;
}

message OrderResponse {
  entity.Order order = 1;
}

message OrderByUserRequest {
  string userId = 1;
}
message OrderByUserResponse {
  repeated entity.Order order = 1;
}

message CreateOrderRequest {
  
}

message CreateOrderResponse {
  entity.Order order = 1;
}
```

```proto
// ./payment.proto
syntax = "proto3";

import "./modules/product.proto";
import "./modules/order.proto";

package payment;

message None {}
```

**Comment by kamilmysliwiec** [comment]
Thanks for reporting @zry656565! Would you like to create a PR?

**Comment by diepnghitinh** [comment]
in 5.7.4
microservice-configuration.interface line 18
package: string; is not working, should be string[] :(

**Comment by zry656565** [comment]
@kamilmysliwiec I created a related PR, please have a look when u'r free.

**Comment by aaskandrew** [comment]
PR https://github.com/nestjs/nest/pull/1774 is about ability to load more then one proto file. 'grpc' package allows this functionality. But this PR about load multiple namespaces in the nest GrpcClient. That can lead to conflicts in grpc services names. More over, problem of this PR resolves with multi level namespaces.

entity.proto
package common.entity;
...

product.proto
package common.product;

GrpcConfig:
    package: 'common',
    protoPath: ['entity.proto', 'product.proto'], <======= PR https://github.com/nestjs/nest/pull/1774

**Comment by chrisenytc** [comment]
@kamilmysliwiec What is the state of this feature request? What can I do to help?

I am using Nestjs to create a microservices ecosystem using gRPC but I am stuck with this issue. I need every gRPC service to run along with a gRPC health check service on the same port using different proto files. 

Example: 
```bash
account.session.v1.proto
grpc.health.v1.proto
```

```protobuf
syntax = "proto3";

package account.session.v1;

service SessionService {
  rpc FindAll(SessionFindAllRequest) returns (SessionFindAllResponse);
}

message Session {
  int32 id = 1;
  string token = 2;
}

message SessionFindAllRequest {}

message SessionFindAllResponse { repeated Session sessions = 1; }
```

```protobuf
syntax = "proto3";

package grpc.health.v1;

message HealthCheckRequest { string service = 1; }

message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
  }
  ServingStatus status = 1;
}

service Health {
  rpc Check(HealthCheckRequest) returns (HealthCheckResponse);

  rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
}
```

I will be using Kubernetes so I need to use terminus to do health checking. I have time to contribute to this because I like Nestjs and want to use it in production at my new startup.

This is the non-breaking solution I am proposing:

```javascript
  app.connectMicroservice({
    transport: Transport.GRPCMultipleServices,
    options: {
      url: 'localhost:5000',
      services: [
        {
          package: 'account.session.v1',
          protoPath: 'account.session.v1.proto',
        },
        {
          package: 'grpc.health.v1',
          protoPath: 'grpc.health.v1.proto',
        },
      ],
    },
  });
```

**Comment by zombiQWERTY** [comment]
+1

**Comment by AnthonyMouallem** [comment]
+1

**Comment by vilkatis** [comment]
+1

**Comment by selginho** [comment]
+1

**Comment by Jeff-Tian** [comment]
+1

**Comment by Jeff-Tian** [comment]
Before the PR get merged, I created a pacakge: `grpc-health` that can enable standard grpc health check to `nestjs grpc application`. I improved the `hero sample app` to show that: https://github.com/Jeff-Tian/nestjs-hero-grpc-sample-with-health-check, the main change to the original sample is here: https://github.com/Jeff-Tian/nestjs-hero-grpc-sample-with-health-check/commit/b76249ccf76d183143d55130825967d3bebe47de

**Comment by kamilmysliwiec** [comment]
Published as 6.11.0. More information here #3418
